lab3/create-cascade.sql

```diff
@@ -0,0 +1,198 @@
+CREATE TABLE positions (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL CHECK (name ~* '^[A-Za-z ]+$'),
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    CHECK (created_at <= updated_at)
+);
+
+CREATE TABLE users (
+    id SERIAL PRIMARY KEY,
+    email VARCHAR(255) NOT NULL UNIQUE CHECK (
+        email ~* '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$'
+    ),
+    password VARCHAR(255) NOT NULL,
+    name VARCHAR(255) NOT NULL CHECK (name !~* '[0-9]'),
+    surname VARCHAR(255) NOT NULL CHECK (surname !~* '[0-9]'),
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    CHECK (created_at <= updated_at)
+);
+
+CREATE TABLE employees (
+    fk_user_id INTEGER PRIMARY KEY,
+    fk_position_id INTEGER NOT NULL,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    CHECK (created_at <= updated_at),
+    FOREIGN KEY (fk_user_id) REFERENCES users(id) ON DELETE CASCADE,
+    FOREIGN KEY (fk_position_id) REFERENCES positions(id) ON DELETE SET NULL
+);
+
+CREATE TABLE addresses (
+    id SERIAL PRIMARY KEY,
+    building_number INTEGER NOT NULL CHECK (building_number > 0),
+    flat_number INTEGER CHECK (flat_number > 0),
+    street VARCHAR(255) NOT NULL,
+    city VARCHAR(255) NOT NULL CHECK (city ~* '^[A-Za-z ]+$'),
+    postcode VARCHAR(255) NOT NULL CHECK (postcode ~* '^[A-Za-z0-9 -]+$'),
+    country VARCHAR(255) NOT NULL CHECK (country IN ('Poland','Estonia')),
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    CHECK (created_at <= updated_at)
+);
+
+CREATE TABLE customers (
+    id SERIAL PRIMARY KEY,
+    fk_user_id INTEGER NOT NULL,
+    fk_address_id INTEGER,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    CHECK (created_at <= updated_at),
+    FOREIGN KEY (fk_user_id) REFERENCES users(id) ON DELETE CASCADE,
+    FOREIGN KEY (fk_address_id) REFERENCES addresses(id) ON DELETE SET NULL
```

Comment:
Tutaj np. usunałem z fk_address_id `NOT NULL`, bo imo nie powinnismy usuwac calego customera jak sie usunie jakis adres co nie🤞😔